initfiles/componentfiles/thor/stop_thor

```diff
@@ -47,23 +47,17 @@ if [ "$#" -lt "2" ] || [ "$2" != "keep_sentinel" ]; then
     sleep 1
 fi
 
-masterpid=`cat ${MASTER_PID_NAME} 2> /dev/null`
-if [ ! -z $masterpid ]; then
-  while :
-  do
-    kill -0 $masterpid >& /dev/null
-    masterRunning=$(( $? == 0 ? 1 : 0 ))
-    if [ 0 == $masterRunning ]; then
-      break
-    fi
-    echo --------------------------
-    echo stopping thormaster $THORMASTER 
-    kill $masterpid >& /dev/null
-    sleep 8
-    kill -9 $masterpid >& /dev/null
-    sleep 1
-  done
+masterpid=$( cat $MASTER_PID_NAME )
+TIMEOUT=8000
+kill $masterpid &>/dev/null
+while [ -d /proc/$masterpid -a $TIMEOUT -gt 0 ]; do
```

Comment:
Same as before, but it feels like we should have a 'checkProcess(<pid>, <procesname>)' utility method in use here instead. To cater for the edge case where the $pid does exist but isn't thormaster.
i.e. check pid exists and proc name is what you expect.

(don't hold up this PR for that suggestion though)